scrapscript.py

```diff
@@ -1005,19 +1038,25 @@ def make_bool(x: bool) -> Object:
     return Symbol("true" if x else "false")
 
 
+def wrap_inferred_number_type(x: Union[int, float]) -> Object:
```

Comment:
My only concern with this is that it relies on the Python language quite a bit. If this is meant to be a reference implementation that can be ported to other languages I'm not sure if we want implement some of this arithmetic type inference from scratch.

Reply:
This is a really good callout

I don't have a good idea for how to do this with few lines of code and I don't love the current code

Reply:
My best idea is to just add a to-do. even if we had one implementation for int and one for float for each binop that's quite a bit of code for some simple operators. And then if you want to be able to multiply/divide/add etc ints with floats it gets even more complex.

I think the alternative is just to not support floats until we have some fancier multiple dispatch..